pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/CreateInvitationEventAction.java

```diff
@@ -0,0 +1,95 @@
+package org.broadinstitute.ddp.model.event;
+
+import java.sql.Timestamp;
+
+import org.broadinstitute.ddp.db.dao.AnswerDao;
+import org.broadinstitute.ddp.db.dao.InvitationDao;
+import org.broadinstitute.ddp.db.dao.InvitationFactory;
+import org.broadinstitute.ddp.db.dto.EventConfigurationDto;
+import org.broadinstitute.ddp.db.dto.InvitationDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.activity.instance.answer.Answer;
+import org.broadinstitute.ddp.model.activity.types.EventTriggerType;
+import org.broadinstitute.ddp.model.invitation.InvitationType;
+import org.broadinstitute.ddp.pex.PexInterpreter;
+import org.broadinstitute.ddp.service.EventService;
+import org.broadinstitute.ddp.util.MiscUtil;
+import org.broadinstitute.ddp.util.TimestampUtil;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateInvitationEventAction extends EventAction {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CreateInvitationEventAction.class);
+
+    private String contactEmailQuestionStableId;
+    private boolean markExistingAsVoided;
+
+    public CreateInvitationEventAction(EventConfiguration eventConfiguration, EventConfigurationDto dto) {
+        this(eventConfiguration, dto.getContactEmailQuestionStableId(), dto.shouldMarkExistingInvitationsAsVoided());
+    }
+
+    public CreateInvitationEventAction(EventConfiguration eventConfiguration,
+                                       String contactEmailQuestionStableId,
+                                       boolean markExistingAsVoided) {
+        super(eventConfiguration, null);
+        this.contactEmailQuestionStableId = contactEmailQuestionStableId;
+        this.markExistingAsVoided = markExistingAsVoided;
+    }
+
+    @Override
+    public void doAction(PexInterpreter interpreter, Handle handle, EventSignal signal) {
+        InvitationCreatedSignal nextSignal = run(handle, signal);
+        EventService.getInstance().processAllActionsForEventSignal(handle, nextSignal);
+    }
+
+    InvitationCreatedSignal run(Handle handle, EventSignal signal) {
+        String contactEmail = fetchContactEmail(handle, signal);
+        if (!MiscUtil.isEmailFormatValid(contactEmail)) {
+            throw new DDPException("Answer contact email is not a valid email");
```

Comment:
Maybe include `contactEmail` in the exception text?

Reply:
Will do in one of the follow-up PRs.